src/compton2/Compton2.cc

```diff
@@ -0,0 +1,475 @@
+  // first eval has all leg moments; others have only the 0th moment
+  num_points_ = num_evals_ + num_leg_moments_ - 1U;
+  UINT egsz = gsz + 1;
+
+  if (false) {
+    std::cout << "DBG num_temperatures_ " << num_temperatures_ << '\n';
+    std::cout << "DBG num_groups_ " << num_groups_ << '\n';
+    std::cout << "DBG num_leg_moments_ " << num_leg_moments_ << '\n';
+    std::cout << "DBG num_evals_ " << num_evals_ << '\n';
+    std::cout << "DBG num_points_ " << num_points_ << '\n';
+    std::cout << "DBG len(first_groups_) " << fgsz << '\n';
+    std::cout << "DBG len(indexes_) " << isz << '\n';
+    std::cout << "DBG len(data_/derivs_) " << dsz << '\n';
+  }
+
+  Ts_.resize(tsz);
+  for (UINT i = 0; i < tsz; ++i)
+    fin.read(reinterpret_cast<char *>(&Ts_[i]), sizeof(Ts_[i]));
+
+  Egs_.resize(egsz);
+  for (UINT i = 0; i < egsz; ++i)
+    fin.read(reinterpret_cast<char *>(&Egs_[i]), sizeof(Egs_[i]));
+
+  first_groups_.resize(fgsz);
+  for (UINT i = 0; i < fgsz; ++i)
+    fin.read(reinterpret_cast<char *>(&first_groups_[i]), sizeof(first_groups_[i]));
+
+  indexes_.resize(isz);
+  for (UINT i = 0; i < isz; ++i)
+    fin.read(reinterpret_cast<char *>(&indexes_[i]), sizeof(indexes_[i]));
+
+  data_.resize(dsz);
+  for (UINT i = 0; i < dsz; ++i)
+    fin.read(reinterpret_cast<char *>(&data_[i]), sizeof(data_[i]));
+
+  derivs_.resize(dsz);
+  for (UINT i = 0; i < dsz; ++i)
+    fin.read(reinterpret_cast<char *>(&derivs_[i]), sizeof(derivs_[i]));
+
+  fin.close();
+
+  if (false) {
+    std::cout << '\n';
+    std::cout << "-------------------------------------------------\n";
+
+    std::cout << "\nDBG Ts";
+    for (UINT i = 0; i < tsz; ++i)
+      std::cout << ", " << Ts_[i] / 510.998;
+    std::cout << '\n';
+
+    std::cout << "\nDBG Egs";
+    for (UINT i = 0; i < egsz; ++i)
+      std::cout << ", " << Egs_[i] / 510.998;
+    std::cout << '\n';
+
+    std::cout << "\nDBG first_groups";
+    for (UINT i = 0; i < fgsz; ++i)
+      std::cout << ", " << first_groups_[i];
+    std::cout << '\n';
+
+    std::cout << "\nDBG indexes";
+    for (UINT i = 0; i < isz; ++i)
+      std::cout << ", " << indexes_[i];
+    std::cout << '\n';
+
+    std::cout << "\nDBG data";
+    for (UINT p = 0; p < num_points_; ++p) {
+      std::cout << "\nPoint " << p << "\n";
+      UINT szp = dsz / num_points_;
+      for (UINT ii = 0; ii < fgsz; ++ii) {
+        UINT istrt = indexes_[ii] + p * szp;
+        UINT iend = indexes_[ii + 1] + p * szp;
+        std::cout << "  index " << ii;
+        for (UINT i = istrt; i < iend; ++i) {
+          std::cout << std::setprecision(12); // << std::scientific;
+          std::cout << ", " << data_[i] / 0.075116337052433;
+        }
+        std::cout << '\n';
+      }
+    }
+    std::cout << std::defaultfloat;
+
+    std::cout << "-------------------------------------------------\n";
+    std::cout << '\n';
+  }
+
+  return 0;
+}
+
+////////////////////////////////////////////////////////////////////////////////
```

Comment:
Please use standard draco formatting.

Reply:
Must have been a copy-paste from somewhere. Is standard draco formatting `/* ---- ... ---- */`?